core/src/autogluon/core/metrics/classification_metrics.py

```diff
@@ -413,7 +413,9 @@ def customized_binary_roc_auc_score(y_true: Union[np.array, pd.Series], y_score:
     tps = np.cumsum(y_true)[threshold_idxs]
     fps = 1 + threshold_idxs - tps
 
-    if len(tps) > 2:
+    if tps.size > 100000:
+        # optimize indices only when there is enough size to justify
+        # this has no impact on the final score
```

Comment:
Ref to original code from sklearn explaining that this is indeed optional: https://github.com/scikit-learn/scikit-learn/blame/f3f51f9b611bf873bd5836748647221480071a87/sklearn/metrics/_ranking.py#L985-L1000

Reply:
You could consider moving this [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) to a well named constant with a comment so that it can be more easily discovered and tuned in the future if necessary. Since this is only used once and there's a related comment below, I leave it up to you if you think it would increase maintainability.

Reply:
Good point! Updated.